### PR TITLE
MM-50966 - In Product Expansion Backend

### PR DESCRIPTION
**Pull request by ConorMacpherson**
#### Summary

These changes extend the current self-hosted purchase confirmation handler to make callouts to CWS in order to expand seats, rather than purchase a new subscription.

#### Ticket Link

[MM-50365](https://mattermost.atlassian.net/browse/MM-50365)

#### Related PRs

[Mattermost Front-End](https://github.com/mattermost/mattermost-server/pull/22658)
[Enterprise](https://github.com/mattermost/enterprise/pull/1406)
[CWS (confirmation)](https://github.com/mattermost/customer-web-server/pull/1148)

#### Release Note

```release-note
NONE
```

[MM-50365]: https://mattermost.atlassian.net/browse/MM-50365?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/channels/api4/hosted_customer.go [hunk 1]
```diff
@@ -325,3 +328,81 @@ func handleSubscribeToNewsletter(c *Context, w http.ResponseWriter, r *http.Requ
...
+
+		if err.Error() == fmt.Sprintf("%d", http.StatusUnprocessableEntity) {
+			c.Err = model.NewAppError(where, "api.cloud.app_error", nil, "", http.StatusUnprocessableEntity).Wrap(err)
+			return
+		}
+		c.Err = model.NewAppError(where, "api.cloud.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		return
+	}
+
+	license, err := c.App.Srv().Platform().SaveLicense([]byte(confirmResponse.License))
```
Comment (AGMETEOR) [inline server/channels/api4/hosted_customer.go:378]: Could rename err here to appErr

Comment (ConorMacpherson) [reply server/channels/api4/hosted_customer.go:378]: Sure thing! :)

Comment (AGMETEOR) [reply server/channels/api4/hosted_customer.go:377]: Looks like you did for the upper one but not here

Comment (ConorMacpherson) [reply server/channels/api4/hosted_customer.go]: My bad! I've now fixed this one.

**Review by AGMETEOR — APPROVED** [review]
LGTM

**Review by neallred — APPROVED** [review]
LGTM!

**Pull request merged**